Sources/TuistCore/DependenciesGraph/DependenciesGraphController.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+/// A protocol that defines an interface to save and load the `DependenciesGraph` using a `graph.json` file.
+public protocol DependenciesGraphControlling {
+    /// Saves the `DependenciesGraph` as `graph.json`.
+    /// - Parameters:
+    ///   - dependenciesGraph: A model that will be saved.
+    ///   - path: Directory whose project's dependencies graph will be saved.
+    func save(_ dependenciesGraph: DependenciesGraph, at path: AbsolutePath) throws
+
+    /// Loads the `DependenciesGraph` from `graph.json` file.
+    /// - Parameter path: Directory whose project's dependencies graph will be loaded.
+    func load(at path: AbsolutePath) throws -> DependenciesGraph
+}
+
+public final class DependenciesGraphController: DependenciesGraphControlling {
+    private let fileHandler: FileHandling
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared
+    ) {
+        self.fileHandler = fileHandler
+    }
+
+    public func save(_ dependenciesGraph: DependenciesGraph, at path: AbsolutePath) throws {
+        let jsonEncoder = JSONEncoder()
+        jsonEncoder.outputFormatting = .prettyPrinted
+
+        let encodedGraph = try jsonEncoder.encode(dependenciesGraph)
+        #warning("WIP: handle force unwrapping better!")
+        let encodedGraphContent = String(data: encodedGraph, encoding: .utf8)!
+        let graphPath = graphPath(at: path)
+
+        try fileHandler.write(encodedGraphContent, path: graphPath, atomically: true)
+    }
+
+    public func load(at path: AbsolutePath) throws -> DependenciesGraph {
+        let graphPath = graphPath(at: path)
```

Comment:
I believe this won't build on Xcode 12.4

Reply:
yes it won't 😅 do we still support Xcode 12.4? what is the lowest version that we should support? 🤔

Reply:
ok, I can see that CI is failed because of that, I will fix it

Reply:
we should be conservative with dropping Xcode versions as not all teams move that quickly (keeping the last few minor versions usually). Plus in this case, Xcode 12.5 is still not supported by Github Actions.